### PR TITLE
Added option to disable a set of AST transformations

### PR DESCRIPTION
**Pull request by azreika**
Added a new command-line option, `--disable-transformers=<TRANSFORMERS>` or `-z<TRANSFORMERS>`, where `<TRANSFORMERS>` is a comma-separated list of transformer names (as shown in the debug report).

Allows the user to disable any provided AST transformation. For example, to disable the InliningTransformer and PartitionBodyLiteralsTransformer transformations, one can use the command:

`souffle test.dl --disable-transformers="PartitionBodyLiteralsTransformer,InliningTransformer"`

or

`souffle test.dl -z"PartitionBodyLiterals,InliningTransformer"`

Should make testing certain configurations much faster.

### CODE REVIEW

**Comment by souffle-lang** [comment]
[65% (-0.13%) vs master 65%](http://plang1.it.usyd.edu.au/jenkins/job/PR-RHEL7-gcc7-Coverage/192/)

**Comment by mmcgr** [comment]
Hey Jenkins,  test this please

File: src/DebugReport.h [hunk 1]
```diff
@@ -129,6 +129,14 @@ class DebugReporter : public MetaTransformer {
         }
     }
 
+    void disableTransformers(const std::set<std::string>& transforms) {
```
Comment (mmcgr) [inline src/DebugReport.h]: Missing an `override` here

**Comment by souffle-lang** [comment]
[65% (-0.15%) vs master 65%](http://plang1.it.usyd.edu.au/jenkins/job/PR-RHEL7-gcc7-Coverage/194/)

**Pull request merged**